code/modules/fish/fishtank.dm

```diff
@@ -0,0 +1,693 @@
+
+/obj/machinery/fishtank/proc/harvest_fish(var/mob/user)
+	if(!fish_count)									//Can't catch non-existant fish!
+		to_chat(usr, "There are no fish in \the [src] to catch!")
+		return
+
+	var/caught_fish
+	caught_fish = input("Select a fish to catch.", "Fishing", caught_fish) in fish_list		//Select a fish from the tank
+	if(caught_fish)
+		var/dead_fish = null
+		dead_fish = fish_items_list[caught_fish]	//Locate the appropriate fish_item for the caught fish
+		if(!dead_fish)								//No fish_item found, possibly due to typo or not being listed. Do nothing.
+			return
+		kill_fish(caught_fish)						//Kill the caught fish from the tank
+		user.visible_message("[user.name] harvests \a [caught_fish] from \the [src].", "You scoop \a [caught_fish] out of \the [src].")
+		new dead_fish(get_turf(user))				//Spawn the appropriate fish_item at the user's feet.
+
+/obj/machinery/fishtank/proc/destroy(var/deconstruct = 0)
+//	var/turf/T = get_turf(src)										//Store the tank's turf for atmos updating after deletion of tank
+	if(!deconstruct)												//Check if we are deconstructing or breaking the tank
+		var/shards_left = shard_count
+		while(shards_left > 0)										//Produce the appropriate number of glass shards
+			new /obj/item/weapon/shard(get_turf(src))
+			shards_left --
+		if(water_level)												//Spill any water that was left in the tank when it broke
+			spill_water()
+	else															//We are deconstructing, make glass sheets instead of shards
+		var/sheets = shard_count + 1								//Deconstructing it salvages all the glass used to build the tank
+		new /obj/item/stack/sheet/glass(get_turf(src), sheets)		//Produce the appropriate number of glass sheets, in a single stack
+	qdel(src)														//qdel the tank and it's contents
+//	T.air_update_turf(1)											//Update the air for the turf, to avoid permanent atmos sealing with wall tanks
+
+/obj/machinery/fishtank/proc/spill_water()
+	switch(tank_type)
+		if("bowl")										//Fishbowl: Wets it's own tile
+			var/turf/T = get_turf(src)
+			if(!istype(T, /turf/simulated)) return
+//			var/turf/simulated/S = T
+	//		S.MakeSlippery()
+		if("tank")										//Fishtank: Wets it's own tile and the 4 adjacent tiles (cardinal directions)
+			var/turf/ST = get_turf(src)
+//			if(istype(ST, /turf/simulated))
+//				var/turf/simulated/ST2 = ST
+//				ST2.MakeSlippery()
+			var/list/L = ST.CardinalTurfs()
+			for(var/turf/T in L)
+				if(!istype(T, /turf/simulated)) continue
+//				var/turf/simulated/S = T
+			//	S.MakeSlippery()
+	//	if("wall")							help me god			//Wall-tank: Wets it's own tile and the surrounding 8 tiles (3x3 square)
+		//	for(var/turf/T in spiral_range_turfs(1, src.loc))  weird ass paradise code yo
+		//	if(!istype(T, /turf/simulated)) continue
+		//	var/turf/simulated/S = T
+	//			S.MakeSlippery()
+
+//////////////////////////////		Note from FalseIncarnate:
+//		EXAMINE PROC		//			This proc is massive, messy, and probably could be handled better.
```

Comment:
This guy was right, this proc is nuts. If you insist on keeping the very verbose result trees, switch statements would be better.